syntax = "proto3";

package test;

message Void {}

message Data {
  string foo = 1;
  int32 bar = 2;
  bool baz = 3;
  string hello = 4;
  int32 world = 5;
  bool lorem = 6;
  string ipsum = 7;
}

message DataBatch {
  repeated Data data = 1;
}

message Response {
  string message = 1;
  Data data = 2;
}

message ResponseBatch {
  string message = 1;
  repeated Data data = 2;
}

service Test {
  rpc Hello(Void) returns (Void) {}
  rpc PostData(Data) returns (Response) {}
  rpc PostDataBatch(DataBatch) returns (ResponseBatch) {}

  // Stream Batch
  rpc PostDataBatchStreamRequest(stream Data) returns (ResponseBatch) {}
  rpc PostDataBatchStreamResponse(DataBatch) returns (stream Response) {}
  rpc PostDataBatchStreamBoth(stream Data) returns (stream Response) {}
}
